opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -36,3 +36,62 @@ pub use sampler::{JaegerRemoteSampler, JaegerRemoteSamplerBuilder};
 
 #[cfg(test)]
 mod runtime_tests;
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use super::*;
+    use crate::testing::trace::InMemorySpanExporterBuilder;
+    use opentelemetry::{
+        trace::{Span, Tracer, TracerProvider as _},
+        KeyValue,
+    };
+
+    #[test]
+    fn tracing_in_span() {
+        // Arrange
+        let exporter = InMemorySpanExporterBuilder::new().build();
+        let provider = TracerProvider::builder()
+            .with_span_processor(SimpleSpanProcessor::new(Box::new(exporter.clone())))
+            .build();
+
+        // Act
+        let tracer = provider.tracer("test_tracer");
+        tracer.in_span("span_name", |_cx| {});
+
+        provider.force_flush();
+
+        // Assert
+        let exported_spans = exporter
+            .get_finished_spans()
+            .expect("Spans are expected to be exported.");
+        assert_eq!(exported_spans.len(), 1);
+        let span = &exported_spans[0];
+        assert_eq!(span.name, "span_name");
+        assert_eq!(span.instrumentation_lib.name, "test_tracer");
+    }
+
+    #[test]
+    fn tracing_tracer_start() {
```

Comment:
Just a nit on the apparent use of `tracing_` prefix in the function names... I'd prefer to rely on the parent module name which is already `trace::tests` and not redundantly add more `tracing_`.

Reply:
oh yes! Will make this part of the next PR where I intend to add more tests, along with extracting common code to helpers.